thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2651,6 +2722,44 @@ class CBucketHandler : public CSimpleInterface, implements IInterface, implement
             peakKeyCount = newPeakKeyCount;
     }
 
+    class CBucketHandlerRowStream : public IRowStream, public CInterface
+    {
+        CBucketHandler * parent;
+        Owned<IRowStream> bucketRowStream;
+        bool stopped;
+        unsigned cur;
+    public:
+        IMPLEMENT_IINTERFACE;
+        CBucketHandlerRowStream(CBucketHandler * _parent) : parent(_parent), cur(0), stopped(false) {};
+        const void * nextRow()
+        {
+            while (!stopped && (cur < parent->numBuckets))
+            {
+                if (!parent->buckets[cur]->isSpilt() && parent->isValidBucket(cur))
+                {
+                    if (!bucketRowStream.get())
+                    {
+                        CBucket * bucket = parent->buckets[cur];
+                        if (bucket->getKeyCount()==0)
+                        {
+                            cur ++;
+                            continue;
+                        }
+                        bucketRowStream.setown(bucket->getRowStream());
+                    }
+                    const void * row = bucketRowStream->nextRow();
```

Comment:
Change will be moot once spilling is active and getNextBucketStream() is added as discussed.